### PR TITLE
enhancement: handling cwv grouping changes

### PR DESCRIPTION
**Pull request by 05rahulsingh**
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Thanks for contributing!

### CODE REVIEW

**Comment by nit23uec** [comment]
I think we should store the cwv metrics along with the url in form-vitals to avoid mapping them again.
```
 "form-vitals": [
    {
      "url": "https://business.adobe.com/se/resources/main.html",
      "formengagement": {
        "desktop:windows": 100
      },
      "pageview": {
        "desktop:windows": 7898
      },
      "cwv": {
           "desktop": {},
            "mobile": {}
        }
    }
]

```

File: test/fixtures/expectedformvitalsdata.json [hunk 1]
```diff
@@ -6,14 +6,21 @@
     "formVitals": [
```
Comment (nit23uec) [inline test/fixtures/expectedformvitalsdata.json:6]: in this data, the urls in cwv seem to be more than those in form vitals. is it expected?

File: src/forms-opportunities/handler.js [hunk 2]
```diff
@@ -34,7 +34,9 @@ export async function formsAuditRunner(auditUrl, context, site) {
 
   const queryResults = await rumAPIClient.queryMulti(FORMS_OPPTY_QUERIES, options);
   const cwvMap = new Map(
-    queryResults.cwv.map((cwv) => [cwv.url, cwv]),
+    queryResults.cwv
+      .filter((cwv) => cwv.type === 'url')
+      .map((cwv) => [cwv.url, cwv]),
```
[the remark this answers is not in the document]
Comment (05rahulsingh) [reply src/forms-opportunities/handler.js:39]: whatever the url's are in a group, that'll be coming also in type:url.
We are matching exact url's, so yep ignoring type:group

File: test/fixtures/expectedformvitalsdata.json (hunk 1 above)
Comment (05rahulsingh) [reply test/fixtures/expectedformvitalsdata.json:6]: yes, form-vitals queries data on different checkpoints and cwv queries data on other set of checkpoints. Yes, it is expected

File: src/forms-opportunities/handler.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ekremney) [reply src/forms-opportunities/handler.js:39]: `type === url` aligns with the previous functionality. All urls with at least one CWV metric will have an entry as previously. 

`type === group` is the new kid in the block which I am not sure if it relates to the form-vitals logic.

**Review by ekremney — APPROVED** [review]

**Comment by 05rahulsingh** [comment]
> I think we should store the cwv metrics along with the url in form-vitals to avoid mapping them again.
> 
> ```
>  "form-vitals": [
>     {
>       "url": "https://business.adobe.com/se/resources/main.html",
>       "formengagement": {
>         "desktop:windows": 100
>       },
>       "pageview": {
>         "desktop:windows": 7898
>       },
>       "cwv": {
>            "desktop": {},
>             "mobile": {}
>         }
>     }
> ]
> ```

would like to keep the structure as it is which we are getting from cwv query.
unless if any major modifications happens again we dont have to do anything at our end, like adding or bifurcating new device type or something like that.
Reporting worker processes this and parses this in a more human friendly manner anyways.

**Review by nit23uec — APPROVED** [review]

**Pull request merged**